File: backend/src/main/java/com/yourorg/appname/service/ProductService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.response.ProductResponse;

import java.util.List;

public interface ProductService {
    List<ProductResponse> getAllProducts(String category);
    List<ProductResponse> getAllProducts(String category, String occasion);
    ProductResponse getProductById(Long id);
}

File: frontend/src/services/productService.js
import apiClient from './apiClient';
import { ENDPOINTS } from '../constants/apiEndpoints';

export const productService = {
  getAllProducts: async (category, occasion) => {
    const params = {};
    if (category && category.toLowerCase() !== 'all') {
      params.category = category;
    }
    if (occasion && occasion.toLowerCase() !== 'all') {
      params.occasion = occasion;
    }
    const response = await apiClient.get(ENDPOINTS.PRODUCTS, { params });
    return response.data.data;
  },

  getProductById: async (id) => {
    const response = await apiClient.get(`${ENDPOINTS.PRODUCTS}/${id}`);
    return response.data.data;
  },
};
